apps/taiga-lumbermill/src/dashboards/crypto/components/swap/swap.component.ts

```diff
@@ -0,0 +1,101 @@
+import {CommonModule} from '@angular/common';
+import {
+    ChangeDetectionStrategy,
+    Component,
+    computed,
+    inject,
+    signal,
+} from '@angular/core';
+import {toSignal} from '@angular/core/rxjs-interop';
+import {FormsModule} from '@angular/forms';
+import {TuiAmountPipe} from '@taiga-ui/addon-commerce';
+import {
+    TuiAppearance,
+    TuiButton,
+    TuiDataList,
+    TuiDropdown,
+    TuiIcon,
+    TuiTitle,
+} from '@taiga-ui/core';
+import {TuiAvatar, TuiChevron, TuiInputInline} from '@taiga-ui/kit';
+import {TuiCardLarge, TuiCell, TuiHeader} from '@taiga-ui/layout';
+
+import type {PricesData} from '../../../../services/crypto.service';
+import {CryptoService} from '../../../../services/crypto.service';
+
+@Component({
+    standalone: true,
+    selector: 'lmb-swap',
+    imports: [
+        CommonModule,
+        FormsModule,
+        TuiAmountPipe,
+        TuiAppearance,
+        TuiAvatar,
+        TuiButton,
+        TuiCardLarge,
+        TuiCell,
+        TuiChevron,
+        TuiDataList,
+        TuiDropdown,
+        TuiHeader,
+        TuiIcon,
+        TuiInputInline,
+        TuiTitle,
+    ],
+    templateUrl: './swap.component.html',
+    styleUrl: './swap.component.less',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SwapComponent {
+    protected readonly cryptoService = inject(CryptoService);
+    protected readonly info = toSignal(this.cryptoService.getTokens());
+    protected readonly priceFrom = computed(() =>
+        this.getPrice(this.info(), this.chosenFrom()),
```

Comment:
this.getPrice(this.tokenFrom())
....

getPrice(token:string) {
  return (this.tokens() ?? []).find(....)
}